backend-tests/tests/test_inventory.py

```diff
@@ -529,14 +826,539 @@ def test_search_v2(self, tenants_users):
+                    "name": "test2",
+                    "terms": [
+                        {
+                            "type": "$in",
+                            "scope": "inventory",
+                            "attribute": "artifact",
+                            "value": ["v2", "v3"],
+                        },
+                        {
+                            "type": "$exists",
+                            "scope": "inventory",
+                            "attribute": "py3",
+                            "value": True,
+                        },
+                        {
+                            "type": "$nin",
+                            "scope": "inventory",
+                            "attribute": "py3",
+                            "value": ["3.5", "3.6"],
+                        },
+                    ],
+                },
+                "status_codes": [201, 200, 200, 200],
+                "result": [
+                    {"id": dev.id, "attributes": dict_to_inventoryattrs(dev.inventory),}
+                    for dev in filter(
+                        lambda dev: dev.inventory["artifact"] in ["v2.0", "v2.1"]
+                        and "py3" in dev.inventory
+                        and dev.inventory["py3"] not in ["3.5", "3.6"],
+                        self.tenant_ent.devices,
+                    )
+                ],
+            },
+            {
+                "name": "Compound filter: $ne -> $lt -> $gte -> $exists",
+                "tenant": self.tenant_ent,
+                "request": {
+                    "name": "test3",
+                    "terms": [
+                        {
+                            "type": "$ne",
+                            "scope": "inventory",
+                            "attribute": "artifact",
+                            "value": "v2.0",
+                        },
+                        {
+                            "type": "$lte",
+                            "scope": "inventory",
+                            "attribute": "idx",
+                            "value": 8,
+                        },
+                        {
+                            "type": "$gt",
+                            "scope": "inventory",
+                            "attribute": "idx",
+                            "value": 1,
+                        },
+                        {
+                            "type": "$exists",
+                            "scope": "inventory",
+                            "attribute": "py3",
+                            "value": False,
+                        },
+                    ],
+                },
+                "status_codes": [201, 200, 200, 200],
+                "result": [
+                    {"id": dev.id, "attributes": dict_to_inventoryattrs(dev.inventory),}
+                    for dev in filter(
+                        lambda dev: "py3" not in dev.inventory
+                        and 2 <= dev.inventory["idx"] < 8
```

Comment:
This is important; the rest of the review is minor stuff.

Reply:
You're right. The test passes because device at index 8 is excluded by another filtering operation, I'm changing this test-case because it doesn't make much sense after I changed the device set.